pkg/agent/config/traffic_encap_mode_test.go

```diff
@@ -0,0 +1,248 @@
+package config
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetTrafficEncapModeFromStr(t *testing.T) {
+	tests := []struct {
+		name    string
+		str     string
+		expBool bool
+		expMode TrafficEncapModeType
+	}{
+		{
+			name:    "encap-mode-valid",
+			str:     "enCap",
+			expBool: true,
+			expMode: 0,
+		},
+		{
+			name:    "no-encap-mode-valid",
+			str:     "Noencap",
+			expBool: true,
+			expMode: 1,
+		},
+		{
+			name:    "hybrid-mode-valid",
+			str:     "Hybrid",
+			expBool: true,
+			expMode: 2,
+		},
+		{
+			name:    "invalid-str",
+			str:     "en cap",
+			expBool: false,
+			expMode: -1,
+		},
+	}
+	for _, tt := range tests {
+		actualBool, actualMode := GetTrafficEncapModeFromStr(tt.str)
+		assert.Equal(t, actualBool, tt.expBool, "GetTrafficEncapModeFromStr not return correct boolean")
```

Comment:
here too we can add the actual and expected values in the message

Reply:
I think assert.Equal will log the actual and expected values once it fails. like this:
```
  traffic_encap_mode_test.go:54: 
        	Error Trace:	traffic_encap_mode_test.go:54
        	Error:      	Not equal: 
        	            	expected: []config.TrafficEncapModeType{0, 1, 2}
        	            	actual  : []config.TrafficEncapModeType{0, 1, 3}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -3,3 +3,3 @@
        	            	  (config.TrafficEncapModeType) NoEncap,
        	            	- (config.TrafficEncapModeType) Hybrid
        	            	+ (config.TrafficEncapModeType) (PANIC=runtime error: index out of range [3] with length 3)3
        	            	 }
        	Test:       	TestGetTrafficEncapModes
        	Messages:   	GetTrafficEncapModes not return correct type mapping
```

Reply:
thanks